run.sh

```diff
@@ -2,6 +2,46 @@
 
 set -o errexit
 
+usage() {
+    echo "deploy                                         -- Deploy SUSE Containerized OpenStack."
+    echo "update_openstack                               -- Update OpenStack deployment."
+    echo "add_openstack_compute                          -- Add OpenStack compute node. User need to provide compute node name in inventory file."
+    echo "remove_openstack_compute <compute-node-name>   -- Remove OpenStack compute node. User need to provide compute node name with option."
+    echo "remove_deployment                              -- Delete all resources related with deployment including images."
+}
+if [ -z "${1:-}" ]; then
+    echo "$0 -h, --help          -- Show help message."
+    exit 1
+fi
+
+OPTIONS="deploy update_openstack add_openstack_compute remove_openstack_compute remove_deployment deploy_network deploy_ses deploy_caasp deploy_ccp_deployer
+enroll_caasp_workers patch_upstream build_images deploy_osh setup_caasp_workers_for_openstack setup_hosts setup_openstack setup_airship
+setup_everything teardown clean_k8s clean_airship_not_images"
+
+action=$1
+validate_options(){
+   isvalid=false
+   for value in ${OPTIONS[@]} ; do
+      if [ $value == $action ]; then
+         isvalid=true
+         break;
+      fi
+   done
+   if [ $isvalid = false ]; then
+      echo "Invalid option, Check --help for valid options"
+      exit 1
+   fi
+   return 0
+}
+
+while true ; do
+    case "$action" in
+      -h|--help) usage ; exit 0 ;;
+       *) validate_options;  break ;;
+    esac
+done
+
+
```

Comment:
nit: I understand why we are doing this early, before any other logic, but this is harder to read than if the functions were in a separate scripts, sourced. For example: source ${scripts_absolute_dir}/pre-flight-checks.sh validate_cli_arguments

Reply:
Done